helm/aladdin-demo/templates/aladdin-demo.deploy.yaml

```diff
@@ -57,10 +60,36 @@ spec:
         volumeMounts:
           - mountPath: /etc/nginx/
             name: {{ .Chart.Name }}-nginx
-{{ end }}
+          # mount html that should contain an index.html file written by the init container
+          - mountPath: /usr/share/nginx/html
+            name: workdir
         envFrom:
           - configMapRef:
               name: {{ .Chart.Name }}
+{{ end }}
+############# init container ################
+## initContainers must run and successfully exit before the pod can start. If it fails, K8s
+## will restart the initContainers until it is successful. 
+## You can have multiple initContainers, they will execute one by one in order
+#############################################
+      initContainers:
```

Comment:
what happens if initContainers is empty? that is if nginx.use = false, and redis.create = false? 
Also unrelated thought, should nginx.use be nginx.create? or is nginx different in redis in terms of "creation"? Minor thought

Reply:
Yeah i actually tried it with empty initContainers, it just ignores that and runs correctly

I guess redis.create refers to whether the app should spin up its own redis instance, but it should always use redis. I'm actually not super sure on how nginx is used by our other services, but here using it also implies creating it, so I think it's good to distinguish the two fields?